contracts/debtAllocators/GenericDebtAllocator.sol

```diff
@@ -0,0 +1,211 @@
+            // Find out by how much.
+            uint256 toPull = Math.min(
+                params.current_debt - targetDebt,
+                // Account for the current liquidity constraints.
+                IVault(_strategy).maxWithdraw(address(_vault))
+            );
+
+            // Check if its over the threshold.
+            if (toPull > config.minimumChange) {
+                // If so return true and the calldata.
+                return (
+                    true,
+                    abi.encodeCall(
+                        _vault.update_debt,
+                        (_strategy, params.current_debt - toPull)
+                    )
+                );
+            }
+        } else {
+            return (false, bytes("No Change Needed"));
+        }
+    }
+
+    /**
+     * @notice Sets a new target debt ratio for a strategy.
+     * @dev A `minimumChange` for that strategy must be set first.
+     * This is to prevent debt to be updated to frequently.
+     *
+     * @param _strategy Address of the strategy to set.
+     * @param _targetRatio Amount in Basis points to allocate.
+     */
+    function setTargetDebtRatio(
+        address _strategy,
+        uint256 _targetRatio
+    ) external onlyGovernance {
+        // Make sure the strategy is added to the vault.
+        require(IVault(vault).strategies(_strategy).activation != 0, "!active");
+        // Make sure a minimumChange has been set.
+        require(configs[_strategy].minimumChange != 0, "!minimum");
+
+        // Get what will be the new total debt ratio.
+        uint256 newDebtRatio = debtRatio -
+            configs[_strategy].targetRatio +
+            _targetRatio;
+
+        // Make sure it is under 100% allocated
+        require(newDebtRatio <= MAX_BPS, "!max");
+
+        // Write to storage.
+        configs[_strategy].targetRatio = _targetRatio;
+        debtRatio = newDebtRatio;
+
+        emit SetTargetDebtRatio(_strategy, _targetRatio, newDebtRatio);
+    }
+
+    /**
+     * @notice Set the minimum change variable for a strategy.
+     * @dev This is the amount of debt that will needed to be
+     * added or pulled for it to trigger an update.
+     *
+     * @param _strategy The address of the strategy to update.
+     * @param _minimumChange The new minimum to set for the strategy.
+     */
+    function setMinimumChange(
+        address _strategy,
+        uint256 _minimumChange
+    ) external onlyGovernance {
+        // Make sure the strategy is added to the vault.
+        require(IVault(vault).strategies(_strategy).activation != 0, "!active");
+
+        // Set the new minimum.
+        configs[_strategy].minimumChange = _minimumChange;
+
+        emit SetMinimumChange(_strategy, _minimumChange);
+    }
+
+    /**
+     * @notice Set the max acceptable base fee.
+     * @dev This defaults to max uint256 and will need to
+     * be set for it to be used.
+     *
+     * @param _maxAcceptableBaseFee The new max base fee.
```

Comment:
maybe add to the comment the base unit that it is expecting. e.g. gwei or wei

Reply:
https://github.com/yearn/vault-periphery/pull/15/commits/21b8ff41c6b5848a617675584c28b81077eb389e